src/openzaak/components/zaken/tests/test_zaak_verlengen.py

```diff
@@ -0,0 +1,399 @@
+                "zaak": f"http://testserver{expected_zaak_url}",
+                "zaakinformatieobjecten": [],
+            },
+        }
+
+        response_data = response.json()
+        self.assertEqual(response_data, expected_response)
+
+    def test_with_empty_zaak(self):
+        content = {"zaak": {}, "status": self.status}
+
+        response = self.client.post(self.url, content)
+
+        self.assertEqual(
+            response.status_code, status.HTTP_400_BAD_REQUEST, response.data
+        )
+
+        error = get_validation_errors(response, "zaak.verlenging")
+        self.assertEqual(error["code"], "required")
+
+    def test_with_empty_status(self):
+        content = {
+            "zaak": {},
+            "status": {},
+        }
+
+        response = self.client.post(self.url, content)
+
+        self.assertEqual(
+            response.status_code, status.HTTP_400_BAD_REQUEST, response.data
+        )
+
+        statustype = get_validation_errors(response, "status.statustype")
+        self.assertEqual(statustype["code"], "required")
+
+        datum = get_validation_errors(response, "status.datumStatusGezet")
+        self.assertEqual(datum["code"], "required")
+
+    def test_with_no_status(self):
+        content = {
+            "zaak": {},
+        }
+
+        response = self.client.post(self.url, content)
+
+        self.assertEqual(
+            response.status_code, status.HTTP_400_BAD_REQUEST, response.data
+        )
+
+    def test_with_no_zaak(self):
+        content = {"status": self.status}
+
+        response = self.client.post(self.url, content)
+
+        self.assertEqual(
+            response.status_code, status.HTTP_400_BAD_REQUEST, response.data
+        )
+
+    def test_zaaktype_statustype_relation(self):
+        content = {
+            "zaak": {},
+            "status": self.status
+            | {"statustype": self.check_for_instance(StatusTypeFactory.create())},
+        }
+
+        response = self.client.post(self.url, content)
+
+        self.assertEqual(
+            response.status_code, status.HTTP_400_BAD_REQUEST, response.data
+        )
+        error = get_validation_errors(response, "status.nonFieldErrors")
+        self.assertEqual(error["code"], "zaaktype-mismatch")
+
+    def test_with_eind_status(self):
+        ResultaatFactory(
+            zaak=self.zaak,
+            resultaattype__brondatum_archiefprocedure_afleidingswijze=BrondatumArchiefprocedureAfleidingswijze.afgehandeld,
+        )
+
+        content = {
+            "zaak": self.verlenging,
+            "status": self.status
+            | {"statustype": f"http://testserver{self.statustype_url_2}"},
+        }
+
+        response = self.client.post(self.url, content)
+
+        self.assertEqual(
+            response.status_code, status.HTTP_400_BAD_REQUEST, response.data
+        )
+        error = get_validation_errors(response, "status.nonFieldErrors")
+        self.assertEqual(error["code"], "eindstatus-not-allowed")
```

Comment:
Got this question from Olha (related to zaak_afsluiten), what happens if you try this with the `geforceerd-bijwerken` scope?